pkg/trace/stats/statsraw.go

```diff
@@ -6,28 +6,82 @@
 package stats
 
 import (
+	"math/rand"
 	"strings"
 
-	"github.com/DataDog/datadog-agent/pkg/trace/stats/quantile"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/sketches-go/ddsketch"
+	"github.com/golang/protobuf/proto"
+)
+
+const (
+	// relativeAccuracy is the value accuracy we have on the percentiles. For example, we can
+	// say that p99 is 100ms +- 1ms
+	relativeAccuracy = 0.01
+	// maxNumBins is the maximum number of bins of the ddSketch we use to store percentiles.
+	// It can affect relative accuracy, but in practice, 2048 bins is enough to have 1% relative accuracy from
+	// 80 micro second to 1 year: http://www.vldb.org/pvldb/vol12/p2195-masson.pdf
+	maxNumBins = 2048
 )
 
 // Most "algorithm" stuff here is tested with stats_test.go as what is important
 // is that the final data, the one with send after a call to Export(), is correct.
 
 type groupedStats struct {
-	topLevel float64
+	// using float64 here to avoid the accumulation of rounding issues.
+	hits            float64
+	topLevelHits    float64
+	errors          float64
+	duration        float64
+	okDistribution  *ddsketch.DDSketch
+	errDistribution *ddsketch.DDSketch
+}
 
-	hits                    float64
-	errors                  float64
-	duration                float64
-	durationDistribution    *quantile.SliceSummary
-	errDurationDistribution *quantile.SliceSummary
+// round a float to an int, uniformly choosing
+// between the lower and upper approximations.
+func round(f float64) uint64 {
+	i := uint64(f)
+	if rand.Float64() < f-float64(i) {
+		i++
+	}
+	return i
+}
+
+func (s *groupedStats) export(k statsKey) (pb.ClientGroupedStats, error) {
+	msg := s.okDistribution.ToProto()
+	okSummary, err := proto.Marshal(msg)
+	if err != nil {
+		return pb.ClientGroupedStats{}, err
+	}
+	msg = s.errDistribution.ToProto()
+	errSummary, err := proto.Marshal(msg)
+	if err != nil {
+		return pb.ClientGroupedStats{}, err
+	}
+	return pb.ClientGroupedStats{
+		Service:        k.aggr.Service,
+		Name:           k.name,
+		Resource:       k.aggr.Resource,
+		HTTPStatusCode: k.aggr.StatusCode,
+		Type:           k.aggr.Type,
+		Hits:           round(s.hits),
+		Errors:         round(s.errors),
+		Duration:       round(s.duration),
+		TopLevelHits:   round(s.topLevelHits),
+		OkSummary:      okSummary,
+		ErrorSummary:   errSummary,
+		Synthetics:     k.aggr.Synthetics,
+	}, nil
 }
 
 func newGroupedStats() *groupedStats {
+	ok, _ := ddsketch.LogCollapsingLowestDenseDDSketch(relativeAccuracy, maxNumBins)
```

Comment:
Is it safe to ignore the second return value?

Reply:
The error is only when relativeAccuracy is outside of [0,1] interval, and it's a constant here.
I'm a bit hesitant to add an error path for an error that can't happen.
What would you do in that case?

Reply:
How about simply `log.Error` and that's it?

Reply:
Sounds good. Updated.